tiger_agent/agent.py

```diff
@@ -223,24 +225,37 @@ async def make_system_prompt(self, ctx: dict[str, Any]) -> str:
             Rendered system prompt string
         """
         tmpl = self.jinja_env.get_template("system_prompt.md")
-        return await tmpl.render_async(**ctx)
+        return await tmpl.render_async(**ctx.model_dump())
 
     @logfire.instrument("make_user_prompt", extract_args=False)
-    async def make_user_prompt(self, ctx: dict[str, Any]) -> str:
+    async def make_user_prompt(self, ctx: AgentResponseContext) -> str | Sequence[UserContent]:
         """Generate user prompt from Jinja2 template.
 
         Renders the 'user_prompt.md' template with the provided context,
         creating a dynamic user prompt that typically contains the actual
         Slack message content and relevant contextual information.
 
+        If the mention contains attached files, downloads them and returns
+        a sequence of UserContent objects (text + binary files) for multimodal
+        processing by the AI agent.
+
         Args:
             ctx: Template context containing event, user, bot info, etc.
 
         Returns:
-            Rendered user prompt string
+            Rendered user prompt string if no files are attached, or a sequence
+            of UserContent objects (text prompt + file contents) if files are present
         """
         tmpl = self.jinja_env.get_template("user_prompt.md")
-        return await tmpl.render_async(**ctx)
+        text_prompt =  await tmpl.render_async(**ctx.model_dump())
+        
+        if ctx.mention.files is None or not len(ctx.mention.files):
+            return text_prompt
+        
+        user_contents = [await download_private_file(file) for file in ctx.mention.files]
```

Comment:
```suggestion
        user_contents: list[UserContent] = [await download_private_file(file) for file in ctx.mention.files]
```
pycharm is complaining about the types here

Reply:
https://github.com/timescale/tiger-agents-for-work/pull/72/commits/2211c8d85ad0b5281343c95c2f696f58aefef2c1